worker/worker.go

```diff
@@ -45,13 +45,30 @@ func (w *Pool) StartWorkers() {
 				batchReadTime := time.Now()
 				//@TODO - Should add integration tests to prove that the worker receives the same message that it produced, on the delivery channel it created
 
+				eventCounts := make(map[string]int)
+				for _, e := range request.GetEvents() {
+					counts, ok := eventCounts[e.Type]
+					if !ok {
+						counts = 0
+					}
+					eventCounts[e.Type] = counts + 1
+				}
+
 				err := w.kafkaProducer.ProduceBulk(request.GetEvents(), deliveryChan)
 				totalErr := 0
+
+				errCounts := make(map[string]int)
 				if err != nil {
-					for _, err := range err.(publisher.BulkError).Errors {
-						if err != nil {
-							logger.Errorf("[worker] Fail to publish message to kafka %v", err)
+					for _, eErr := range err.(publisher.BulkError).Errors {
```

Comment:
Just as a suggestion if we are doing some changes related to BulkError next time maybe we can explore multi error https://pkg.go.dev/github.com/hashicorp/go-multierror#:~:text=go%2Dmultierror%20is%20a%20package,list%20and%20access%20the%20errors.

Reply:
This code was already there, I just did a small change to it. Maybe we can create a separate PR for multi error.